exec/cnex/lib/sqlite.c

```diff
@@ -0,0 +1,382 @@
+#ifdef _WIN32
+#define _CRT_SECURE_NO_WARNINGS
+#endif
+#include "sqlite.h"
+
+#include <inttypes.h>
+
+#include "assert.h"
+#include "cell.h"
+#include "exec.h"
+#include "nstring.h"
+#include "number.h"
+#include "object.h"
+#include "stack.h"
+
+#include "sqlite3.h"
+
+
+
+static Dictionary *Cursors;
+
+void sqlite_initModule(void)
+{
+    Cursors = dictionary_createDictionary();
+}
+
+void sqlite_shutdownModule(void)
+{
+    dictionary_freeDictionary(Cursors);
```

Comment:
Do you want to actually close any cursors that might be left in this dictionary, or not worry about it?

Reply:
I wasn't sure if I should worry about it, or make it the programmers responsibility.  I think the code will exception on close if you have a cursor that is left open.  I'll test it, try it and see how it runs.